runtime/interp.c

```diff
@@ -72,9 +72,9 @@ sp is a local copy of the global variable caml_extern_sp. */
 #define Restore_after_gc \
   { accu = sp[0]; env = sp[1]; sp += 2; }
 #define Setup_for_c_call \
-  { saved_pc = pc; *--sp = env; caml_extern_sp = sp; }
+  { sp-=2; sp[0] = env; sp[1] = (value)pc; caml_extern_sp = sp; }
```

Comment:
Looks good.  You might get better backtraces by saving `pc + 1` (i.e. the PC of the instruction following the `C_CALL`) instead of `pc`.  The debug event, if any, points to "after" the `C_CALL`, so `event_for_location` should find it more easily if the saved pc also points to "after" the `C_CALL`.  All this despite the tolerance built in `event_for_location`.

Reply:
Also: the house style is `sp -= 2`, not `sp-=2`, which looks like your space bar got jammed.